spring-cloud-gcp-data-firestore/src/test/java/com/google/cloud/spring/data/firestore/it/FirestoreIntegrationTests.java

```diff
@@ -273,6 +273,14 @@ public void writeReadDeleteTest() {
 		assertThat(this.firestoreTemplate.deleteAll(User.class).block()).isEqualTo(2);
 		assertThat(this.firestoreTemplate.findAll(User.class).collectList().block()).isEmpty();
 
+		alice.setUpdateTime(null);
+		this.firestoreTemplate.save(alice).block();
+		bob.setUpdateTime(null);
+		this.firestoreTemplate.save(bob).block();
+
+		this.firestoreTemplate.deleteById(Flux.just("Bob", "Saitama", "Alice"), User.class).block();
+		assertThat(this.firestoreTemplate.count(User.class).block()).isZero();
```

Comment:
Instead of blocking, try using `StepVerifier` here -- it will both subscribe to the publisher stream, and allow you to assert the stream content with `.expectNext()`.

Reply:
Ah, thanks for the suggestion. `StepVerifier` are quite powerful in testing reactive stream!